### ISSUE TITLE
Improve data collection script

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/kata-containers#3047]
We have a data collection script that we ask users to run in the [issue template](https://github.com/kata-containers/.github/blob/main/.github/ISSUE_TEMPLATE/bug_report.md):

- https://github.com/kata-containers/kata-containers/blob/main/src/runtime/data/kata-collect-data.sh.in

However, it hasn't been touched in a while. There are some things we should or could consider doing:

- Review and simplify

  Consider removing all the Kata 1.x information.
- Consider turning it into a "normal" shell script (`*.sh`) by removing the need to expand all the `@some_name@` variables.
- Make the script capture any missing third-party tool versions.
- Potentially remove all the `$problem_pattern` code (since the `level=warn` and `level=error` should cover all those cases).
- Add a CLI option to call `show_image_details()` and/or `show_initrd_details()` so users can run those functions easily.
- Add in some sort of way to identify the *type* of [Kata installation](https://github.com/kata-containers/kata-containers/tree/main/docs/install). Specifically, make it easy to identify if the installation was:
  - build from source.
  - manually installed.
  - installed using `kata-deploy`.

- Show clearly if a newer version is available

  By having the script run `kata-runtime kata-check --check-version-only` (it shouldn't exit on error in case it's run on a disconnected system though!)

**Comment by jodh-intel** [comment]
If anyone is looking for a fun shell project for over Christmas, this might be it... ?!? :santa: :mrs_claus: :sled: :christmas_tree: :snowman:

**Comment by jodh-intel** [comment]
Related: #3128.

**Comment by bookinabox** [comment]
> Potentially remove all the $problem_pattern code (since the level=warn and level=error should cover all those cases).

What guarantee is there that only logs on those levels will be relevant?

**Comment by bookinabox** [comment]
Also, I'm not sure where to get started for determining the installation type and missing third-party tools.

**Comment by wainersm** [comment]
Hi @jodh-intel ! I'm helping @bookinabox implementing this issue.

Regarding "Consider removing all the Kata 1.x information", here are some stuffs I think we should remove:

  * In `show_package_versions()` it should not get information about "cc-oci-runtime|cc-runtime|runv" because they are Clear Containers packages?
  * Remove the entries for docker and podman in `show_container_mgr_details()` as they are no longer supported on kata 2.0?

Anything else?

**Comment by jodh-intel** [comment]
@wainersm - that sounds good, in addition to what I've already written above (throttler for example). Another consideration of course is any changes that might be necessary for Kata 3.x, but that could be dealt with on another PR (and probably only once the changes mentioned here have been synced to the `runtime-rs` branch to allow a new `runtime-rs` PR to be raised for the Kata 3.x collection script changes ;)
